ifrs17-template/Constants/Sharepoint.ipynb

```diff
@@ -37,17 +37,52 @@
         {
             "cell_type": "code",
             "source": [
-                "public class Sharepoint{",
-                "\n    public const string Sites  = \"sites/XXX\"; //Sharepoint sites",
-                "\n    public const string Tenant = \"systemorph.sharepoint.com\";             //Sharepoint tenant",
-                "\n    public const string Root = \"Dev/\";                                    //Dev main directory",
-                "\n    public const string PathToExport = Root + \"ExportFilesTest/\"          //Target export directory ",
-                "\n    public const string PathToImport = Root + \"ImportFiles/\"",
+                "public record SharePointConnectionConfiguration(string Sites = \"sites/XXX\", string Tenant = \"systemorph.sharepoint.com\", string Root = \"Dev/\", bool isApplicationType = false);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static void SetDefaults(this Systemorph.SharePoint.SharePointVariable sharePointVariable, SharePointConnectionConfiguration sharePointConnectionConfiguration) {",
+                "\n    if(sharePointConnectionConfiguration.isApplicationType)",
+                "\n        sharePointVariable.SetDefault(x => x.WithSitesSelected().Site(sharePointConnectionConfiguration.Sites, sharePointConnectionConfiguration.Tenant));",
+                "\n    else",
+                "\n        sharePointVariable.SetDefault(x => x.Site(sharePointConnectionConfiguration.Sites, sharePointConnectionConfiguration.Tenant));",
```

Comment:
what's this method .WithSitesSelected() doing?
A markdown would be useful.

Reply:
Yep, right, it is in fact related to your next comment. I added the appropriate markdowns. This in fact run the registration into the the SharePoint by the app rather than by the user. It does not agree with the settings of the Systemorph SharePoint, but might be required by the settings of certain external stakeholders. Therefore the option is there.